packages/ui-app/src/InputNumber.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { TranslationFunction } from 'i18next';
+import { RawParam$OnChange } from './Params/types';
+import { BareProps } from './types';
+
+import BN from 'bn.js';
+import React from 'react';
+
+import isValidBalance from './util/isValidBalance';
+import { keydown } from './util/keyboard';
+import classes from './util/classes';
+import Input from './Input';
+import { KEYS_ALLOWED, KEYS_PRE } from './constants';
+
+type Props = BareProps & {
+  defaultValue?: string,
+  isError?: boolean,
+  label?: any,
+  maxLength?: number,
+  onChange?: RawParam$OnChange | undefined,
+  placeholder?: string,
+  t: TranslationFunction,
+  withLabel?: boolean
+};
+
+type State = {
+  error?: React.ReactNode,
+  info?: React.ReactNode,
+  warn?: React.ReactNode,
+  isPreKeyDown: boolean
+};
+
+class InputNumber extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = {
+      error: '',
```

Comment:
(nitpick) Since these are marked as optional, we actually don't need to assign values to them here.